client/lib/gsuite/index.js

```diff
@@ -87,12 +88,13 @@ function getEligibleGSuiteDomain( selectedDomainName, domains ) {
 	if ( selectedDomainName && canDomainAddGSuite( selectedDomainName ) ) {
 		return selectedDomainName;
 	}
-
 	const supportedDomains = getGSuiteSupportedDomains( domains );
-
 	const primaryDomain = find( supportedDomains, 'isPrimary' );
-
-	return get( primaryDomain, 'name', '' );
+	const primaryDomainName = get( primaryDomain, 'name', '' );
+	if ( ! primaryDomainName && supportedDomains.length ) {
+		return get( supportedDomains[ 0 ], 'name', '' );
+	}
+	return primaryDomainName;
```

Comment:
Another approach could be to sort the list of supported domains by primary domain, then just return the first element of this list.